src/template.rs

```diff
@@ -76,6 +80,14 @@ pub struct Template {
     context: Context,
 }
 
+// this token needs to be exactly 4 chars; this is temporarily used when formatting
+// the template into a comment regex with the correct column width that can match
+// the [year] against /[\d]{4}/
+const INTERMEDIATE_YEAR_TOKEN: &'static str = "@YR@";
```

Comment:
I know this is will not be immediately obvious, so let me give some context.

Creating the regex to match headers with arbitrary dates is a multistep process:

1. resolve the raw template from either config or SPDX
2. identify replacement tokens
3. replace tokens with configured values, _except for the year_
4. convert to a comment-string, wrapping at the configured columns
5. replace the year token with a 4-digit regex pattern
6. convert the string to a regex instance

It is between steps 3 and 4 that necessitates this token. We need to reliably calculate the column width when converting to a comment, but the replacement tokens for year can be of arbitrary width because of inconsistencies.

Moreover, the replacement token for year isn't guaranteed to be 4 characters long either, but the actual year must be (since we don't care about the middle ages :) and will be long dead before the year 10000 to care about licenses).

So, we need to replace the replacement token with a string that is exactly 4 chars long to wrap to the correct width, but also be unique enough so that we can replace it with a regex pattern and not collide with any text that might be in the license text already.

So, choosing this weird string is not 100% bulletproof since a user can enter anything for the copyright template, but is probably good enough.

If you have a better approach, please feel free to update this! It was the best compromise of simplicity and completeness/correctness I could think of at the time.

Reply:
This LGTM can you add this section to the existing code comment:
> chars long to wrap to the correct width, but also be unique enough so that we can replace it with a regex pattern and not collide with any text that might be in the license text already.

Reply:
done!